xls/examples/lfsr.x

```diff
@@ -0,0 +1,518 @@
+////////////////////////////////////////////////////////////////////////////////
+// Linear-feedback shift register (LFSR)
+// A template function to compute the next value of an LFSR counter.
+// Taps can be specified to have varying periods for a given number of bits.
+//
+// For example, lfsr(u5:4, u5:0b10100) computes the value that comes after 4
+// in a 5-bit LFSR with taps on bits 2 and 4.
+////////////////////////////////////////////////////////////////////////////////
+
+fn lfsr<BIT_WIDTH: u32>(current_value: uN[BIT_WIDTH], tap_mask: uN[BIT_WIDTH]) -> uN[BIT_WIDTH] {
+	// Compute the new bit from the taps
+	let new_bit = for (index, xor_bit): (u32, u1) in range(u32:0, BIT_WIDTH) {
+		if tap_mask[index+:u1] == u1:0 {
+			xor_bit
+		} else {
+			xor_bit ^ current_value[index+:u1]
+		}
+	} (u1:0);
+
+	// Kick the high bit and insert the new bit
+	current_value[0:{(BIT_WIDTH - u32:1) as s32}] ++ new_bit
```

Comment:
curious if https://google.github.io/xls/dslx_reference/#width-slice would make things easier there?

Reply:
I used the width slice instead of the cast.